server/repository/src/db_diesel/item.rs

```diff
@@ -102,6 +103,34 @@ impl<'a> ItemRepository<'a> {
         Ok(query.count().get_result(&self.connection.connection)?)
     }
 
+    pub fn count_no_stock(
```

Comment:
I don't think this is needed, please see comment about using just ItemStats, but if this is needed, we should just add filter and count to StockOnHand repository:  https://github.com/openmsupply/open-msupply/compare/927-item-stats...Stock-on-hand-filter-has-stock

I think it's a good idea to use the power of database rather then manually filtering in rust, unless it's very hard to do in sql